### DISCUSSION TITLE
Redirects within Poem OpenAPI

### DISCUSSION THREAD

**Discussion by ChillFish8** [poem-web/poem#225] [Q&A] [answered]
Hello again 👋 

Im aware we have `Redirect` within poem but this doesn't seem to be supported for poem openapi? Is there anyway we can get around that?

**Comment by sunli829** [comment, marked as answer]
You can define a `301` response with the `ApiResponse` macro. 🙂

```rust
#[derive(ApiResponse)]
enum MyResponse {
    #[oai(status = "301")]
    Redirect(#[oai(header = "Location")] String)
}
```

**Reply by ChillFish8** [reply]
Man that macro really does think of everything huh 😄
